devlib/target.py

```diff
@@ -1785,6 +1785,7 @@ def __init__(self,
                                             is_container=is_container,
                                             max_async=max_async)
         self.package_data_directory = package_data_directory
+        self.external_storage_app_dir = self.path.join(self.external_storage, 'Android', 'data')
```

Comment:
I would suggest using the property decorator for this similar to the existing `external_storage` implementation to defer the evaluation.